eman2/protocols/protocol_tomo_template_match.py

```diff
@@ -0,0 +1,201 @@
+from pyworkflow.em.convert import ImageHandler
+from pyworkflow.em.data import Volume
+from pyworkflow.utils.path import moveFile, cleanPath, makePath
+
+import eman2
+from eman2.convert import loadJson
+
+from tomo.protocols import ProtTomoPicking
+from tomo.objects import Coordinate3D, SetOfCoordinates3D
+
+
+class EmanProtTempMatch(ProtTomoPicking):
+    """
+    This protocol wraps *e2spt_tempmatch.py* EMAN2 program.
+
+    It will perform a sweep of an initial volume against a tomogram
+    to find correlation peaks and extract the corresponding subtomogram
+    coordinates
+
+    """
+
+    _label = 'template matching'
+
+    def __init__(self, **args):
+        ProtTomoPicking.__init__(self, **args)
+
+    # --------------------------- DEFINE param functions ----------------------
+
+    def _defineParams(self, form):
+        form.addSection(label='Input')
+        form.addParam('inputSet', PointerParam,
+                      pointerClass='SetOfTomograms',
+                      label="Input tomograms", important=True,
+                      help='Specify tomograms containing reference-like particles\n'
+                           'to be extracted')
+        form.addParam('ref', PointerParam, pointerClass="Volume",
+                      label='Reference volume',
+                      help='Specify a 3D volume')
+        form.addParam('nptcl', IntParam, default=500,
+                      label='Number of particles',
+                      help='Maximum number of particles')
+        form.addParam('dthr', FloatParam, default=-1,
+                      label='Distance threshold',
+                      help='')
+        form.addParam('vthr', FloatParam, default=2.0,
+                      label='Value threshold',
+                      help='')
+        form.addParam('delta', FloatParam, default=30.0,
+                      label='Delta angle',
+                      help='')
+        form.addParam('sym', StringParam, default='c1',
+                      label='Point-group symmetry',
+                      help='')
+        form.addParam('boxSize', IntParam, default=-1, label='Box size')
+
+        form.addParallelSection(threads=1, mpi=1)
+
+    # --------------------------- INSERT steps functions ----------------------
+
+    def _insertAllSteps(self):
+        self._insertFunctionStep('tempMatchStep')
+        self._insertFunctionStep("createOutputStep")
+
+
+    # --------------------------- STEPS functions -----------------------------
+
+    def tempMatchStep(self):
+
+        if self.boxSize.get()==-1:
+            self.box = self.ref.get().getDim()[0]
+        else:
+            self.box = self.boxSize.get()
+
+        volFile = os.path.abspath(self.ref.get().getFileName())
+        params = ""
+
+        for tomo in self.inputSet.get():
+            params = params + " %s" % tomo.getFileName()
+
+        params = params + " --reference=%s --nptcl=%d --dthr=%f --vthr=%f --delta=%f --sym=%s --rmedge --rmgold --boxsz=%d" %(
```

Comment:
Very minor comment but this line is not PEP8. Do you have the PEP8 validation in pycharm?